packages/js/src/core/sign-in.ts

```diff
@@ -0,0 +1,47 @@
+import qs from 'query-string';
+
+import { withDefaultScopes } from '../utils/scope';
+
+export type SignInUriParameters = {
+  authorizationEndpoint: string;
+  clientId: string;
+  redirectUri: string;
+  codeChallenge: string;
+  state: string;
+  scope?: string | string[];
+  resource?: string[];
+};
+
+enum ResponseType {
+  code = 'code',
+}
+
+export const generateSignInUri = ({
+  authorizationEndpoint,
+  clientId,
+  redirectUri,
+  codeChallenge,
+  state,
+  scope,
+  resource,
+}: SignInUriParameters) => {
+  const queryStringWithoutResource = qs.stringify({
+    client_id: clientId,
+    redirect_uri: [redirectUri],
+    code_challenge: codeChallenge,
+    code_challenge_method: 'S256',
+    state,
+    response_type: ResponseType.code,
+    prompt: 'consent',
```

Comment:
Do we need to have enums for `code_challenge_method` and `prompt`?

Reply:
added local constants for now and will extract them as enum when needed.